src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParseException.cs

```diff
@@ -6,22 +6,32 @@
 namespace System.Text.RegularExpressions
 {
     [Serializable]
-    internal sealed class RegexParseException : ArgumentException
+    public sealed class RegexParseException : ArgumentException
     {
-        private readonly RegexParseError _error; // tests access this via private reflection
-
         /// <summary>Gets the error that happened during parsing.</summary>
-        public RegexParseError Error => _error;
+        public RegexParseError Error { get; }
 
         /// <summary>Gets the offset in the supplied pattern.</summary>
         public int Offset { get; }
 
-        public RegexParseException(RegexParseError error, int offset, string message) : base(message)
+        public RegexParseException(RegexParseError error, int offset)
```

Comment:
Should we have a default message?

Reply:
That's a good point, I can check the message-creating code and make sure it falls back to `Unknown` with something like "Unexpected regular expression parsing error.".